py-polars/polars/expr/datetime.py

```diff
@@ -390,6 +390,98 @@ def round(self, every: str | dt.timedelta | IntoExprColumn) -> Expr:
         every = parse_into_expression(every, str_as_lit=True)
         return wrap_expr(self._pyexpr.dt_round(every))
 
+    def replace(
+        self,
+        *,
+        year: int | IntoExpr | None = None,
+        month: int | IntoExpr | None = None,
+        day: int | IntoExpr | None = None,
+        hour: int | IntoExpr | None = None,
+        minute: int | IntoExpr | None = None,
+        second: int | IntoExpr | None = None,
+        microsecond: int | IntoExpr | None = None,
+        ambiguous: Ambiguous | Expr = "raise",
+    ) -> Expr:
+        """
+        Replace time unit.
+
+        Parameters
+        ----------
+        year
+            Column or literal.
+        month
+            Column or literal, ranging from 1-12.
+        day
+            Column or literal, ranging from 1-31.
+        hour
+            Column or literal, ranging from 0-23.
+        minute
+            Column or literal, ranging from 0-59.
+        second
+            Column or literal, ranging from 0-59.
+        microsecond
+            Column or literal, ranging from 0-999999.
+        ambiguous
+            Determine how to deal with ambiguous datetimes:
+
+            - `'raise'` (default): raise
+            - `'earliest'`: use the earliest datetime
+            - `'latest'`: use the latest datetime
+            - `'null'`: set to null
+
+        Returns
+        -------
+        Expr
+            Expression of data type :class:`Date` or :class:`Datetime` with the
+            specified time units replaced.
+
+        Examples
+        --------
+        >>> from datetime import datetime
+        >>> df = pl.DataFrame(
+        ...     {
+        ...         "date": [datetime(2024, 1, 1), datetime(2024, 1, 2)],
+        ...         "year": [2022, 2016],
+        ...         "month": [1, 2],
+        ...         "day": [4, 5],
+        ...         "hour": [12, 13],
+        ...         "minute": [15, 30],
+        ...     }
+        ... )
+        >>> df.with_columns(
+        ...     pl.col("date").dt.replace(
+        ...         year="year", month="month", day="day", hour="hour", minute="minute"
+        ...     )
```

Comment:
i'd suggest keeping the example simpler - just replace 1 or 2 fields, but most importantly, alias the result, so that it shows before vs after